metastore/src/java/org/apache/hadoop/hive/metastore/MetaStoreUtils.java

```diff
@@ -1472,22 +1472,25 @@ public static FieldSchema getFieldSchemaFromTypeInfo(String fieldName,
         "generated by TypeInfoUtils.getFieldSchemaFromTypeInfo");
   }
 
+  private static boolean isTableTypeExternal(Table table) {
+    String tableType = table.getTableType();
+    return tableType != null && tableType.equals(TableType.EXTERNAL_TABLE.name());
```

Comment:
Another way to do this is to return:
TableType.EXTERNAL_TABLE.name().equals(table.getTableType())